src/shared/Microsoft.Git.CredentialManager/EnvironmentBase.cs

```diff
@@ -43,6 +43,18 @@ public interface IEnvironment
         /// <param name="path">First instance of the found executable program.</param>
         /// <returns>True if the executable was found, false otherwise.</returns>
         bool TryLocateExecutable(string program, out string path);
+
+        /// <summary>
+        /// Create a process ready to start, with redirected streams.
+        /// </summary>
+        /// <param name="path">Absolute file path of executable or command to start.</param>
+        /// <param name="args">Command line arguments to pass to executable.</param>
+        /// <param name="useShellExecute">
+        /// True to resolve <paramref name="path"/> using the OS shell, false to use as an absolute file path.
+        /// </param>
+        /// <param name="workingDirectory">Working directory for the new process.</param>
+        /// <returns><see cref="Process"/> object ready to start.</returns>
+        Process CreateProcess(string path, string args, bool useShellExecute, string workingDirectory);
```

Comment:
Do you ever need to pass a custom environment block?  That is, a custom set of environment variables for the child process to inherit.  Like a custom PATH or debugging/test flags or etc.

I'm not saying you actually need it right now, but rather just wondering if we might at some point.

Reply:
Wait, I see you have that in a later class. I think.  I think it is hard to figure out the class hierarchy in a PR....

Reply:
> Do you ever need to pass a custom environment block?

No. We don't use this (at least at the moment). We just rely on default inheritance behaviour. Should we need this, then yes we can add this in the signature. I'd rather not do that now.

Aside: Windows <--> WSL environment inheritance works differently, and you have to be explicit in which envars are shared across the boundary. The `WSLENV` envar defines which envars are shared, and if there is any path-style transformations. See https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/

> I think it is hard to figure out the class hierarchy in a PR

The hierarchy is:

```text
IEnvironment <------- EnvironmentBase <--inherit-- PosixEnvironment
                             ^
                             |
                          inherit
                             |
                      WindowsEnvironment
 
```